include/wsrep/server_state.hpp

```diff
@@ -287,18 +288,42 @@ namespace wsrep
          * Load WSRep provider.
          *
          * @param provider WSRep provider library to be loaded.
-         * @param provider_options Provider specific options string
-         *        to be passed for provider during initialization.
+         * @param provider_options_cb Callback to get provider options.
+         *                            The function to be called must be
+         *                            idempotent.
          * @param services Application defined services passed to
          *                 the provider.
          *
          * @return Zero on success, non-zero on error.
          */
         int load_provider(const std::string& provider,
-                          const std::string& provider_options,
+                          const std::function<std::string(provider_options&)>&,
```

Comment:
Add parameter name `provider_options_cb` to make declaration match the documentation above.

Reply:
fixed